dm_control/utils/inverse_kinematics.py

```diff
@@ -207,8 +231,18 @@ def qpos_from_site_pose(physics,
       # Update `physics.qpos`, taking quaternions into account.
       mjlib.mj_integratePos(physics.model.ptr, physics.data.qpos, update_nv, 1)
 
+      # clip joint angles to their respective limits
+      if len(sites_names) > 1:
+        joint_names = physics.named.data.qpos.axes.row.names
+        limited_joints = joint_names[1:]  # ignore root joint
```

Comment:
You probably want to explicitly check which joints are actually limited using `jnt_limited`.